UniqueFilenameClass/UniqueFilename.cs

```diff
@@ -0,0 +1,33 @@
+﻿using MyConsoleApp.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyConsoleApp.UniqueFilenameClass
+{
+    public class UniqueFilename : IUniqueFilename
```

Comment:
Aside from the class name, which you'd likely change if you update the interface name, this code looks solid.